### PR TITLE
CI Rework

### PR DESCRIPTION
**Pull request by labbott**

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -0,0 +1,64 @@
+name: release
+on:
+  workflow_call:
+    inputs:
+      build:
+        description: "Machine we are releasing for"
+        required: true
+        type: string
+      app_name:
+        description: "Machine we are releasing for"
```
Comment (steveklabnik) [inline .github/workflows/release.yml:10]: these descriptions probably should be updated to be accurate, obviously not an issue right this moment

File: .github/workflows/dist.yml [hunk 2]
```diff
@@ -32,7 +31,7 @@ jobs:
             app_name: lpc55xpresso
             app_toml: app/lpc55xpresso/app.toml
             target: thumbv8m.main-none-eabihf
-            image: [a, b]
+            image: "a, b"
```
Comment (steveklabnik) [inline .github/workflows/dist.yml:34]: i am interested in this change, I'm not sure why it's being done

**Review by steveklabnik — COMMENTED** [review]
Neat, I haven't used the feature where workflows let you call other workflows yet, but this makes sense to me!

File: .github/workflows/dist.yml (hunk 2 above)
Comment (labbott) [reply .github/workflows/dist.yml:34]: Reusable workflows can only handle inputs of boolean, number or string https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#onworkflow_callinputsinput_idtype so I converted this to a comma separated string. It turns out we were already doing this in the build anyway.

Comment (steveklabnik) [reply .github/workflows/dist.yml:34]: Ah, interesting, thank you!

**Review by steveklabnik — APPROVED** [review]

**Pull request merged**